src/test/arith_uint256_tests.cpp

```diff
@@ -97,27 +91,17 @@ BOOST_AUTO_TEST_CASE( basics ) // constructors, equality, inequality
     }
     BOOST_CHECK(ZeroL == (OneL << 256));
 
-    // String Constructor and Copy Constructor
-    BOOST_CHECK(arith_uint256S("0x" + R1L.ToString()) == R1L);
```

Comment:
Some of these were reverted since it seemed to us that it's testing prefixes and whitespaces primarily, see https://github.com/bitcoin/bitcoin/pull/30773#discussion_r1740942391

Reply:
> is not trivially clear to me that the UintToArith256 tests can be removed

+1 for restoring them in that case

Reply:
I'm definitely happy to put the existing tests back in if it makes review easier, but I want to make sure I understand your concern, so let me try to summarize it:

Prior to https://github.com/bitcoin/bitcoin/commit/facf629ce8ff1d1f6d254dde4e89b5018f8df60e, to construct a `arith_uint256` from a hex string, one would use the `arith_uint256` string constructor, and as such we had tests covering that constructor. Since https://github.com/bitcoin/bitcoin/commit/facf629ce8ff1d1f6d254dde4e89b5018f8df60e that constructor is removed, and now the de facto approach to construct a `arith_uint256` from a hex string is to first construct a `uint256` from a hex string, and then convert it with `UintToArith256()`. So, rather than removing the string construction unit tests entirely, you kept them by testing this new "roundabout" approach instead, since that's how these functions are used in practice.

Is it fair to summarize that your concern is that even though the tests I remove here are theoretically not necessary since we should have equivalent unit test coverage on the 2 underlying functions (`uint256::FromHex()` and `UintToArith256()`), you'd rather keep them because ensuring that the unit test coverage is indeed equivalent is burdensome and makes this PR harder to review?

If my summary is correct, and it seems the main concern is about `UintToArith256()` test coverage (and not so much `uint256::FromHex()` coverage, I think my latest force-push should address your concerns by vastly expanding (and at the same time simplifying) the `conversion` tests in 32505699afa4b249bc6a8233cbaca97315461f25?

Reply:
> and the tests run so fast that in the case where they are redundant, no one should notice.

Absolutely, the cleanup is to avoid having tests that have unclear purpose, making them easier to maintain going forward. Performance is not the goal here.

I think it's impossible for this PR to not have significant changes to the unit tests, so I feel like since the review is happening anyway the test improvements I've introduced here make sense and add little overhead?

That said, I'm unsure if you're waiting for me to revert things or if you're happy with the current approach, so just to make it explicit could you please clarify if you'd prefer:
1) the `conversion` test changes to be reverted
2) the "string constructor" tests to be re-added
3) moving the `conversion` and `operator_with_self` to `arith_uint256_tests.cpp` to be reverted (not explicitly discussed, but I'm inferring from your message that this might be a concern)

Reply:
>  so I feel like since the review is happening anyway the test improvements I've introduced here make sense and add little overhead?

Agree, let's leave the campground cleaner than we found it.

Reply:
Okay, I've re-added the "string constructor" tests. I think they can be removed later, but that's obviously not critical either. Thanks for explaining your concerns.